DirectProgramming/C++SYCL_FPGA/Tutorials/Features/max_interleaving/README.md

```diff
@@ -62,22 +68,59 @@ The sample illustrates the following important concepts.
+#### Detailed Explanation
+Consider the following pipelined doubly nested loops: 
+```cpp
+float temp_r[kSize] = ...;
+float temp_a[kSize * kSize] = ...;
 
-See the following example:
-
-```
-L1: for (size_t i = 0; i < kSize; i++) {
-  L2: for (size_t j = 0; j < kSize; j++) {
-        temp_r[j] = SomethingComplicated(temp_a[i][j], temp_r[j]);
-  }
-  temp_r[i] += temp_b[i];
+for (int i = kSize - 1; i >= 0; i--) {
+   [[intel::max_interleaving(0 or 1)]] // Controls if interleaving is enabled or disabled on the loop below
+   for (int j = kSize - 1; j >= 0; j--) {
+      temp_r[i] += SomethingComplicated(temp_a[i * kSize + j], temp_r[i]);
+   }
 }
 ```
 
-In this loop nest, pipelined iterations of L1 are serialized across L2 to preserve the data dependency on the array variable 'temp_r'. This means that one invocation of the `j` loop can be executing at any time; therefore, no dynamic interleaving of iterations from different invocations of the `j` loop can occur. By default, the compiler will generate a datapath that includes the capacity to run multiple interleaved iterations simultaneously. Since the data dependency prevents dynamic interleaving, the resources spent on an interleaving-capable datapath are wasted. Applying the `max_interleaving` attribute with an argument of `1` will instruct the compiler generate a datapath that restricts the interleaving capacity to a single `j` loop invocation.
+Notice how `temp_r[i]` is a loop carried dependency in the inner loop. **Crucially, the dependency is with respect to the inner loop and *not* the outer loop.** The same `temp_r[i]` is updated in every inner iteration, but a different element of `temp_r` is updated in different outer iterations. As a result of the loop carried dependency with respect to the inner loop, the II of the inner loop will be very high as a new iteration of the inner loop cannot be invoked until the previous one finishes. 
+
+Without interleaving, this is what the pipelined registers of the inner loop will look like, for II of the inner loop = 5 and II of the outer loop = 1. Each cell shows the (i, j) iteration that is currently in that stage of the inner loop:
+
+| Cycle | Stage 1 | Stage 2 | Stage 3 | Stage 4 | Stage 5 |
```

Comment:
Please align the tables in the textual form for people reading the README in a text editor